modules/v2x/v2x_proxy/app/v2x_proxy.cc

```diff
@@ -13,147 +13,200 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  *****************************************************************************/
-
 /**
  * @file v2x_proxy.cc
  * @brief define v2x proxy class
  */
-
 #include "modules/v2x/v2x_proxy/app/v2x_proxy.h"
 
+#include <cmath>
+#include <cstdio>
+#include <ostream>
+#include <vector>
+
+#include "modules/v2x/common/v2x_proxy_gflags.h"
+
 namespace apollo {
 namespace v2x {
+using ::apollo::canbus::ChassisDetail;
+using ::apollo::v2x::CarStatus;
+using ::apollo::v2x::obu::OBURsi;
+using ::apollo::v2x::obu::OBUTrafficLightData;
+using ::apollo::v2x::obu::StatusResponse;
+
+V2xProxy::~V2xProxy() {
+  if (recv_thread_ != nullptr && recv_thread_->joinable()) {
+    recv_thread_->join();
+  }
+  if (planning_thread_ != nullptr && planning_thread_->joinable()) {
+    planning_thread_->join();
+  }
+  // rsi_thread_->join();
+}
 
-using apollo::localization::LocalizationEstimate;
-
-V2xProxy::V2xProxy()
-    : node_(cyber::CreateNode("v2x_proxy")), init_flag_(false) {
+V2xProxy::V2xProxy(std::shared_ptr<::apollo::hdmap::HDMap> hdmap)
+    : node_(::apollo::cyber::CreateNode("v2x_proxy")) {
   if (node_ == nullptr) {
     AFATAL << "Create v2x proxy node failed.";
     exit(1);
   }
-
-  auto x2v_trafficlight_timer_period = ceil(
-      (1.0 / static_cast<int>(FLAGS_x2v_trafficlight_timer_frequency)) * 1000);
-  x2v_trafficlight_timer_.reset(
-      new cyber::Timer(static_cast<uint32_t>(x2v_trafficlight_timer_period),
-                       [this]() { this->OnX2vTrafficLightTimer(); }, false));
-  auto v2x_carstatus_timer_period = ceil(
+  internal_ = std::make_shared<InternalData>();
+  hdmap_ = std::make_shared<::apollo::hdmap::HDMap>();
+  if (0 != hdmap_->LoadMapFromFile(apollo::hdmap::BaseMapFile())) {
```

Comment:
Don't do many things on one call.
Use a meaningful variable to refer to `apollo::hdmap::BaseMapFile()`.

This PR is too big. Let me take a break.